source/uwp/Renderer/lib/XamlBuilder.cpp

```diff
@@ -1126,6 +1127,253 @@ namespace AdaptiveCards { namespace Rendering { namespace Uwp
+
+        ComPtr<ABI::Windows::Data::Xml::Dom::IXmlNode> hrefNode;
+        RETURN_IF_FAILED(attributeMap->GetNamedItem(HStringReference(L"href").Get(), &hrefNode));
+
+        if (hrefNode == nullptr)
+        {
+            return E_INVALIDARG;
+        }
+
+        HString href;
+        RETURN_IF_FAILED(GetTextFromXmlNode(hrefNode.Get(), href.GetAddressOf()));
+
+        ComPtr<IUriRuntimeClassFactory> uriActivationFactory;
+        RETURN_IF_FAILED(GetActivationFactory(
+            HStringReference(RuntimeClass_Windows_Foundation_Uri).Get(),
+            &uriActivationFactory));
+
+        ComPtr<IUriRuntimeClass> uri;
+        RETURN_IF_FAILED(uriActivationFactory->CreateUri(href.Get(), uri.GetAddressOf()));
+
+        ComPtr<ABI::Windows::UI::Xaml::Documents::IHyperlink> hyperlink = XamlHelpers::CreateXamlClass<ABI::Windows::UI::Xaml::Documents::IHyperlink>(HStringReference(RuntimeClass_Windows_UI_Xaml_Documents_Hyperlink));
+        RETURN_IF_FAILED(hyperlink->put_NavigateUri(uri.Get()));
+
+        ComPtr<ABI::Windows::UI::Xaml::Documents::ISpan> hyperlinkAsSpan;
+        RETURN_IF_FAILED(hyperlink.As(&hyperlinkAsSpan));
+
+        ComPtr<IVector<ABI::Windows::UI::Xaml::Documents::Inline*>> hyperlinkInlines;
+        RETURN_IF_FAILED(hyperlinkAsSpan->get_Inlines(hyperlinkInlines.GetAddressOf()));
+
+        ComPtr<ABI::Windows::UI::Xaml::Documents::IRun> run;
+        RETURN_IF_FAILED(XamlRunFromXmlNode(node, &run));
+
+        ComPtr<ABI::Windows::UI::Xaml::Documents::IInline> runAsInline;
+        RETURN_IF_FAILED(run.As(&runAsInline));
+
+        RETURN_IF_FAILED(hyperlinkInlines->Append(runAsInline.Get()));
+
+        ComPtr<ABI::Windows::UI::Xaml::Documents::IInline> hyperLinkAsInline;
+        RETURN_IF_FAILED(hyperlink.As(&hyperLinkAsInline));
+        RETURN_IF_FAILED(inlines->Append(hyperLinkAsInline.Get()));
+
+        return S_OK;
+    }
+
+    HRESULT AddTextInline(
+        IAdaptiveRenderContext* renderContext,
+        ABI::Windows::Data::Xml::Dom::IXmlNode* node,
+        BOOL isBold,
+        BOOL isItalic,
+        IVector<ABI::Windows::UI::Xaml::Documents::Inline*>* inlines)
+    {
+        ComPtr<ABI::Windows::UI::Xaml::Documents::IRun> run;
+        RETURN_IF_FAILED(XamlRunFromXmlNode(node, &run));
+
+        ComPtr<ABI::Windows::UI::Xaml::Documents::ITextElement> runAsTextElement;
+        RETURN_IF_FAILED(run.As(&runAsTextElement));
+
+        if (isBold)
+        {
+            ComPtr<IAdaptiveHostConfig> hostConfig;
+            RETURN_IF_FAILED(renderContext->get_HostConfig(&hostConfig));
+
+            ComPtr<IAdaptiveFontWeightsConfig> fontWeightsConfig;
+            RETURN_IF_FAILED(hostConfig->get_FontWeights(&fontWeightsConfig));
+
+            ABI::Windows::UI::Text::FontWeight boldFontWeight;
+            RETURN_IF_FAILED(fontWeightsConfig->get_Bolder(&boldFontWeight.Weight));
+
+            RETURN_IF_FAILED(runAsTextElement->put_FontWeight(boldFontWeight));
```

Comment:
just to be sure, <bold><bold>"hello"</bold></bold> will work, right? adding two fontweight won't add the weight, right? thanks for entertaining silly question.

Reply:
Assuming you mean "strong", that example makes "hello" be in bold, which i think is what's supposed to happen.

What doesn't work though is <strong> <em> "hello" </em></strong> which should be both bold and italic and only turns out bold...

---
In reply to: [167377400](https://github.com/Microsoft/AdaptiveCards/pull/1113#discussion_r167377400) [](ancestors = 167377400)

Reply:
Reworked this a bit to make sure that the recursion works properly and my second example now works.

---
In reply to: [167381058](https://github.com/Microsoft/AdaptiveCards/pull/1113#discussion_r167381058) [](ancestors = 167381058,167377400)